### PR TITLE
ScenarioRunner: move the building of the speedy machine

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/ScenarioRunner.scala [hunk 1]
```diff
@@ -26,16 +26,10 @@ import scala.util.{Failure, Success, Try}
 /** Speedy scenario runner that uses the reference ledger.
   *
   * @constructor Creates a runner using an instance of [[Speedy.Machine]].
-  * @param partyNameMangler allows to amend party names defined in scenarios,
```
[the remark this answers is not in the document]
Comment (akrmn) [reply daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/ScenarioRunner.scala:29]: :fire:
=

File: daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala [hunk 2]
```diff
@@ -209,7 +209,7 @@ object Repl {
           compiledPackages,
           expr,
         )
-      (machine, new ScenarioRunner(machine, seed).run)
+      (machine, ScenarioRunner.run(() => machine, seed))
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala:212]: You are write. We need more work if we want to fix LF repl:
However
-  this PR tackles the problem for the Scenario service (see `Context.interpretScenario`
- The LF repl is an internal tool, we can survive without handle properly out of memory error.

[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala:212]: we used it to run the tests in `/daml-lf/test`.

[the remark this answers is not in the document]
Comment (akrmn) [reply daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala:212]: yeah I'm sure I've used it before and probably even touched the code but I didn't _know_ about it 😅

**Pull request merged**